File: Algorithm_448_Find All Numbers Disappeared in an Array.cpp
#include <vector>
#include <cmath>
using namespace std;
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        /*unordered_map<int,int> map;
        vector<int> result;
        for (int i=0;i<nums.size();i++) map[nums[i]]++;
        for (int i=1;i<=nums.size();i++)
        {
            if (map.find(i) == map.end()) result.push_back(i);
        }
        return result;*/
        vector<int> result;
        for (int i=0;i<nums.size();i++)
        {
            int index=abs(nums[i])-1;
            nums[index]=nums[index]>0?-nums[index]:nums[index];
        }
        for (int i=0;i<nums.size();i++) if (nums[i]>0) result.push_back(i+1);
        return result;
    }
};

File: Algorithm_448_Find All Numbers Disappeared in an Array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        result = []
        for i in range(len(nums)):
            idx = abs(nums[i])
            if nums[idx - 1] >  0:
                nums[idx - 1] = -nums[idx - 1]
        for i in range(len(nums)):
            if nums[i] > 0: # i + 1 never doesn't exist in nums
                result.append(i + 1)
        return result